transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -119,6 +119,20 @@ static void netty_epoll_native_eventFdWrite(JNIEnv* env, jclass clazz, jint fd,
     jint eventFD = eventfd_write(fd, (eventfd_t) value);
 
     if (eventFD < 0) {
+        // We need to read before we can write again, let's try to read and then write again and if this
+        // fails we will bail out.
+        //
+        // See http://man7.org/linux/man-pages/man2/eventfd.2.html.
+        if (errno == EAGAIN) {
+            uint64_t val;
+
+            // Just ignore the return value as we will re-issue the write after it and will report its return value.
+            eventfd_read(fd, &val);
+
+            if (eventfd_write(fd, (eventfd_t) value) != 0) {
```

Comment:
This is racy, and it's easy to reproduce by using large numbers for `value`.  I was able to cause this to fail using Long.MAX_VALUE / 5 in the test.  

The race is that event_fd read can fail if another thread beats it to doing the read.  The same race can occur again when doing the write, which can also fail with eagain.

This needs to be done in a loop.  (#9191 helps a lot, but it isn't water tight.)

Reply:
roughly:

```
do {
  err = write(value);
  if (err == 0) {
    return;
  }
  if (errno == EAGAIN) {
    err = read();
    if (err == 0 || errno == EAGAIN) {
      continue;
    }
  }
  throw;
} while (true);
```

Reply:
@carl-mastrangelo sure I can do this... that said we always use `1L` as value in our code.

Reply:
@carl-mastrangelo PTAL again